PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/filesystem/AzureStorageVirtualFileSystem.java

```diff
@@ -0,0 +1,15 @@
+package com.microsoft.azure.hdinsight.spark.ui.filesystem;
+
+import com.intellij.openapi.vfs.impl.http.HttpFileSystemBase;
+import com.microsoft.azure.hdinsight.common.logger.ILogger;
+import com.sun.istack.NotNull;
+import rx.Observable;
+
+public abstract class AzureStorageVirtualFileSystem extends HttpFileSystemBase implements ILogger {
+    public AzureStorageVirtualFileSystem(String protocol) {
+        super(protocol);
+    }
+
+    @NotNull
+    public abstract Observable<AzureStorageVirtualFile> getAzureStorageVirtualFiles(String rootPath);
```

Comment:
The method should be a part of `VirtualFile`, such as `getChildren()`

Reply:
we use rootPath as the param to set the root in the vfs.As for getChilldren, the parent path is already known.

---
In reply to: [285431147](https://github.com/microsoft/azure-tools-for-java/pull/3124#discussion_r285431147) [](ancestors = 285431147)

Reply:
parentFiles.put(root.getPath(), root);

        fileSystem.getAzureStorageVirtualFiles(rootPath)
                .doOnNext(file -> {
                    if (file.isDirectory()) {
                        parentFiles.put(file.getPath(), file);
                    }